### PR TITLE
relay, syncer(dm): rename UUIDSuffix to RelaySubDirSuffix and others

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #4287 

### What is changed and how it works?

make some concepts more clear. Most changes are renaming.

#### Tests 

 - No code

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/pkg/streamer/hub.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ var (
 // RelayLogInfo represents information for relay log.
 type RelayLogInfo struct {
 	TaskName   string
-	UUID       string
```
Comment (D3Hunter) [inline dm/pkg/streamer/hub.go:34]: maybe rename `UUIDSuffix` below too

File: dm/pkg/utils/relay.go [hunk 2]
```diff
@@ -67,18 +67,19 @@ func ParseUUIDIndex(indexPath string) ([]string, error) {
...
 // SuffixIntToStr convert int-represented suffix to string-represented.
+// TODO: assign RelaySubDirSuffix a type and implement Stringer.
 func SuffixIntToStr(id int) string {
 	return fmt.Sprintf("%06d", id)
 }
 
-// ParseSuffixForUUID parses UUID (with suffix) to (UUID without suffix, suffix) pair.
-func ParseSuffixForUUID(uuid string) (string, int, error) {
+// ParseSuffixFromRelaySubDir parses relay log subdirectory name to (server UUID, RelaySubDirSuffix) pair.
+func ParseSuffixFromRelaySubDir(uuid string) (string, int, error) {
```
Comment (D3Hunter) [inline dm/pkg/utils/relay.go:82]: maybe `ParseRelaySubDir`

File: dm/syncer/streamer_controller.go [hunk 3]
```diff
@@ -285,21 +285,15 @@ func (c *StreamerController) GetEvent(tctx *tcontext.Context) (event *replicatio
 	switch ev := event.Event.(type) {
 	case *replication.RotateEvent:
 		// if is local binlog, binlog's name contain uuid information, need to save it
-		// if is remote binlog, need to add uuid information in binlog's name
-		c.Lock()
-		containUUID := c.setUUIDIfExists(string(ev.NextLogName))
-		uuidSuffix := c.uuidSuffix
-		c.Unlock()
-
-		if !containUUID {
```
Comment (D3Hunter) [inline dm/syncer/streamer_controller.go:294]: we may switch 2 remote when using relay and meet error, so we need to add suffix if there's ever set `c.uuidSuffix`.

the comments before should be:
```
// if is local binlog but switch to remote on error, need to add uuid information in binlog's name
```

the lock can be removed though, only `getEvent` routine is using it.

File: dm/syncer/streamer_controller.go [hunk 4]
```diff
@@ -342,15 +337,10 @@ func (c *StreamerController) IsClosed() bool {
...
-func (c *StreamerController) setUUIDIfExists(filename string) bool {
-	_, uuidSuffix, _, err := binlog.SplitFilenameWithUUIDSuffix(filename)
-	if err != nil {
-		// don't contain uuid in position's name
-		return false
-	}
-
-	c.uuidSuffix = uuidSuffix
-	return true
+func (c *StreamerController) extractRelaySubDirSuffix(filename string) string {
```
Comment (D3Hunter) [inline dm/syncer/streamer_controller.go:340]: seems we can inline it

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

**Comment by lance6716** [comment]
ping @GMHDBJD

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**